### DISCUSSION TITLE
Creating OSM notes

### DISCUSSION THREAD

**Discussion by angoca** [labexp/osmtracker-android#604] [Ideas]
It will be very useful to integrate OSM tracker with notes, in a way that instead of inserting the POIs as part of the GPX trace, it creates notes in OSM. In this way, other mappers can start mapping the observations. This prevents downloading the comments from the mobile phone to the computer.

**Comment by mfrasca** [comment]
As it is now, OSMTracker only adds waypoints along the track, this would not let us put the Note (in OSM sense) at an arbitrary position. To add this functionality, we need to allow OSMTracker to add nodes at a position we choose by hand and not by GPS reading. This would be quite a nice addition, and I would advocate for keeping such node/note/waypoints in a separate table in the OSMTracker database, so that we would see them on on the map regardless which track we're revising.

The management of node/note/waypoints, separating private ones from OSM ones would require quite a bit of work, from design to implementation.

Maybe we could start by defining/implementing private waypoints / personal notes, then see how to go further from there?

**Comment by Binnette** [comment]
I think "creating OSM notes" does not fit current OSMTracker usage. May be consider using OsmAnd instead.

**Comment by jamescr** [comment]
After a conversation with @angoca during State of the Map Latam 2025, we can consider including a boolean config option like "upload text notes as OSM notes."  If the option is ON, we record the notes separate from the trace and upload them as OSM notes; otherwise, the text notes are recorded as part of the trace (as it works now). 

Upload to OSM could be done with [osmapi](https://github.com/westnordost/osmapi) (we include the lib when we [switch to oauth2](https://github.com/labexp/osmtracker-android/issues/359)).

Will convert this issue to a discussion.

**Comment by jamescr** [comment]
cc: @JoseAndresVargas

**Reply by angoca** [reply]
This could give some ideas about how to create notes.

https://www.openstreetmap.org/user/aselnigu/diary/407570

**Comment by JoseAndresVargas** [comment]
Hello everyone, I've started working in this feature. 
* Progress: https://github.com/labexp/osmtracker-android/pull/621

In the following  days I'll be finishing the remaining work specified in the PR. I would appreciate your feedback!

**Comment by jamescr** [comment]
Included in the [nightly](https://github.com/labexp/osmtracker-android/releases/tag/nightly) version. (Help is needed with documentation on the [wiki](https://github.com/labexp/osmtracker-android/wiki)).
